awesome.c

```diff
@@ -901,20 +913,44 @@ main(int argc, char **argv)
 
     ewmh_init_lua();
 
+    /* Parse and run configuration file before adding the screens */
+    if (globalconf.no_auto_screen)
+    {
+        /* Disable automatic screen creation, awful.screen has a fallback */
+        globalconf.ignore_screens = true;
+
+        if(!luaA_parserc(&xdg, confpath))
+            fatal("couldn't find any rc file");
+    }
+
+    screen_emit_scanning();
+
     /* init screens information */
     screen_scan();
 
-    /* Parse and run configuration file */
-    if (!luaA_parserc(&xdg, confpath))
+    /* Parse and run configuration file after adding the screens */
+    if (((!globalconf.no_auto_screen) && !luaA_parserc(&xdg, confpath)))
         fatal("couldn't find any rc file");
 
     p_delete(&confpath);
 
     xdgWipeHandle(&xdg);
 
+    screen_emit_scanned();
+
+    /* Exit if the user doesn't read the instructions properly */
+    if (globalconf.no_auto_screen && !globalconf.screens.len)
+        fatal("When -m/--screen is set to \"off\", you **must** create a "
+              "screen object before or inside the screen \"scanned\" "
```

Comment:
Well, it does not really have to be inside of `scanned`. I could e.g. also create one when the config is being loaded. However, I guess it makes sense to be helpful-clear instead of pendantic-clear here, right?

Reply:
I think this is clear enough. It has the "before or inside" part. This is a user facing message. Being  pedantic is counter intuitive.

The point of documentation and errors is to make the user *less* confused if he doesn't know the ins and outs of the code. If someone knows the code, they don't need the doc or error messages.